### DISCUSSION TITLE
AttributeError: 'GlobalStorage' object has no attribute 'num_sampled_nodes' with `NeighborLoader`

### DISCUSSION THREAD

**Discussion by ClaudMor** [pyg-team/pytorch_geometric#8192] [Q&A]
Hello,

The following code:

```python
import torch_geometric as pyg

data_path = "path/to/Reddit/data/folder"

reddit = pyg.datasets.Reddit(root = data_path)



train_loader = pyg.loader.NeighborLoader(   reddit[0], 
                                            num_neighbors = [ 25, 10 ],
                                            batch_size = 512,
                                            shuffle = True, 
                                            input_nodes =  reddit[0].train_mask,
                                            ) 


batch = next(iter(train_loader))
batch.num_sampled_nodes
```

Gives:

```bash
[...]
AttributeError: 'GlobalStorage' object has no attribute 'num_sampled_nodes'
```
I'm currently working on a Windows machine.

I guess the reason for the error is that `NeighborLoader` requires pyg-lib to report the `num_sampled_nodes` (see [code](https://github.com/pyg-team/pytorch_geometric/blob/62599d19811b0e41af9f81476ca4ef3e7e2d92e7/torch_geometric/sampler/neighbor_sampler.py#L314) for `NeighborSampler`, which is internally called by `NeighborLoader`), and pyg-lib is not yet supported on Windows. Is it correct? Is there any way around it while pyg-lib is still unavailable on Windows?

* PyG version: 2.4.0
* PyTorch version: 2.1.0
* OS: WIndows 10
* Python version:  3.11.5
* CUDA/cuDNN version: 12.1 (CUDA)
* How you installed PyTorch and PyG (`conda`, `pip`, source):  I installed Pytorch with Conda and PyG with pip
* Any other relevant information (*e.g.*, version of `torch-scatter`): Since I'm on a WIndows machine, I have torch_sparse installed but not pyg-lib

Thank you.

**Comment by akihironitta** [comment]
@ClaudMor I think you can maybe build `pyg-lib` from source. @rusty1s might be able to clarify on this, but I see some commits for Windows build https://github.com/search?q=repo%3Apyg-team%2Fpyg-lib+windows&type=commits, so I assume it's not impossible for you to build it from source on your own.

**Reply by rusty1s** [reply]
I haven't digged to deep into Windows support yet for `pyg-lib`, sorry. In theory, this should work just fine, and you may already find success via installing it from source. If you can confirm that it works for you, I can try to revisit Windows wheels afterwards.
